File: src/dsa/arrays/inserting/unsorted_inserting_manually.ts
const unsorted_array = [12, 2, 45, 23, 56, 8];
const new_element_2 = 20;
const new_unsorted_array = insertElement(unsorted_array, new_element_2);

console.log(unsorted_array, 'Array antes da inserção');
console.log(new_unsorted_array, 'Array depois da inserção');

function insertElement(arr: number[], element: number) {
    const new_unsorted_array = new Array(arr.length + 1);

    for (let i = 0; i< arr.length; i++) {
        new_unsorted_array[i] = arr[i];
    }

    new_unsorted_array[arr.length] = element;

    return new_unsorted_array;
}

File: src/dsa/arrays/inserting/unsorted_inserting_manually.rs
pub fn run() {
    let unsorted_array: Vec<i32> = vec![8, 12, 2, 34, 26];
    let new_element: i32 = 10;

    println!("Array inicial: {:?}", unsorted_array);

    let new_unsorted_array: Vec<i32> = insert_element(unsorted_array, new_element);

    println!("Array depois da inserção: {:?}", new_unsorted_array);
}

fn insert_element(arr: Vec<i32>, element: i32) -> Vec<i32> {
    let mut new_arr: Vec<i32> = Vec::with_capacity(arr.len() + 1);

    for &item in arr.iter() {
        new_arr.push(item);
    }

    new_arr.push(element);

    new_arr
}
